compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -436,13 +465,28 @@ ofInterestRule opts = do
                     let doc = formatScenarioResult world res
                     when (vr `Set.member` openVRs) $
                         sendEvent $ vrChangedNotification vr doc
+                let vrScenarioNames = Set.fromList $ fmap (vrScenarioName . fst) (concat $ maybeToList mbVrs)
+                forM_ (openVRsForFile file) $ \ovr -> do
+                    when (not $ vrScenarioName ovr `Set.member` vrScenarioNames) $
+                        sendEvent $ vrNoteSetNotification ovr $ LF.scenarioNotInFileNote $
+                        T.pack $ fromNormalizedFilePath file
 
         -- We don’t always have a scenario service (e.g., damlc compile)
         -- so only run scenarios if we have one.
         let shouldRunScenarios = isJust envScenarioService
+
+        let getDalfOrNotifyOpenVrs file = do
+            mbDalf <- getDalf file
+            _ <- if isNothing mbDalf
```

Comment:
`when (isNothing mbDalf)` should work nicely here. I would also change this to not return the `mbDalf` since you are immediately throwing it away via `void`.

Reply:
Thanks!